### ISSUE TITLE
Unable to build ScanTailor-Advanced on Windows

### ISSUE THREAD

**Issue by ableeker** [ScanTailor-Advanced/scantailor-advanced#83]
Building ScanTailor-Advanced on Windows has been quite hard. The building instructions first explain how to build the dependencies, it took a few tries, but in the end I've been able to build these with MinGW64. Then the instructions tell you how to build ST, however I've had no luck with that so far. When I generate the cmake files, I get the following a number of times:

```CMake Error at src/math/tests/CMakeLists.txt:7 (add_executable):
  Target "math_tests" links to target "Boost::unit_test_framework" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
```
This ends with the following:

```-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

However, cmake files have been generated, so when I consider this a warning, and try to build ST, I get the following:

```In file included from E:\Source\ScanTailor\scantailor-advanced\src\math\LinearSolver.cpp:4:0:
E:\Source\ScanTailor\scantailor-advanced\src\math\LinearSolver.h:9:10: fatal error: boost/scoped_array.hpp: No such file or directory
 #include <boost/scoped_array.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
mingw32-make[2]: *** [src\math\CMakeFiles\math.dir\build.make:96: src/math/CMakeFiles/math.dir/LinearSolver.cpp.obj] Error 1
mingw32-make[2]: *** Waiting for unfinished jobs....
In file included from E:\Source\ScanTailor\scantailor-advanced\src\math\LinearFunction.h:9:0,
                 from E:\Source\ScanTailor\scantailor-advanced\src\math\LinearFunction.cpp:4:
E:/Source/ScanTailor/scantailor-advanced/src/foundation/VecT.h:7:10: fatal error: boost/scoped_array.hpp: No such file or directory
 #include <boost/scoped_array.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
mingw32-make[2]: *** [src\math\CMakeFiles\math.dir\build.make:180: src/math/CMakeFiles/math.dir/LinearFunction.cpp.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:649: src/math/CMakeFiles/math.dir/all] Error 2
mingw32-make: *** [Makefile:181: all] Error 2
```
Am I missing something?

**Comment by zvezdochiot** [comment]
@ableeker say:
>Am I missing something?

You have not installed the `boost` required for compilation.

**Comment by ableeker** [comment]
I have compiled Boost, and the other dependencies. That wasn't easy either. It's available in Libs, but when I tried to compile SA, I still got these errors.

**Comment by zvezdochiot** [comment]
@ableeker say:
>It's available in Libs

`Boost` is not available for STA builds. This is clearly indicated to you in the build error message.

**Comment by ableeker** [comment]
I've been able to compile STA eventually. I haven't changed anything, but I've decided to delete everything, and start all over again. I haven't done anything different, but this time, Libs did compile, and this time were complete, and as a result STA did compile successfully at last.

**Comment by ableeker** [comment]
Closed, because the proposed fix has been implemented by PR #94.
